### ISSUE TITLE
Config Presets

### ISSUE THREAD

**Issue by rgant** [eslint-stylistic/eslint-stylistic#55]
Could you add a "all" (like "eslint:all") and "recommended" (like "eslint:recommended") rule configurations?

My philosophy is that I turn on all rules, then disable the ones that are not for my project. That way I don't have to keep track of new rules being added or changed.

**Comment by rgant** [comment]
Would be especially nice for turning off the js rules that have ts versions.

**Comment by ljharb** [comment]
My experience from maintaining plugins is that any included configuration is actually a bad choice, because adding enabled rules to it - even an “all” - is technically a breaking change. It’d be fine to provide configs, but imo they should be provided in a separate package.

**Comment by rgant** [comment]
One dev's breaking change is another dev's code quality shortcut.

eslint and @typescript-eslint both provide all & recommended configs. No one is forced to use them if they don't want to. A separate package just means that they will end up [out of sync](https://github.com/stylelint-scss/stylelint-config-recommended-scss/issues/251) in my experience. 

Having to figure out which JS rules to disable for TS is not something every developer should have to figure out individually.

I just spent an hour trying to figure out how to migrate my configs for eslint-stylistic. Would have saved me a bunch of time if there was a config here.

**Comment by ljharb** [comment]
I agree that it’s convenient, and i don’t see why they’d get out of sync - that’s trivial to automate and enforce.

**Comment by antfu** [comment]
I am not sure about naming them `all` and `recommended`. But maybe something like `prettierlike` where we can try to mimic prettier's default preference. It would take some effort to check them one by one to create such config, we will have it in our backlog

**Comment by Namchee** [comment]
To achieve prettier-like configuration, I think we can start by analyzing rules that are disabled by [eslint-config-prettier](https://github.com/prettier/eslint-config-prettier/blob/main/index.js) since they should be handled by prettier.

**Comment by kevintyj** [comment]
I have taken a look at prettier configs and put the default preferences here:
The @stylistic is the related @stylistic rules

## Prettier default [options](https://prettier.io/docs/en/options.html)

- Print Width: **80** 
  - @stylistic/max-len: **`{'code': 80}`**
- Tab Width: **2** 
  - @stylistic/max-len: **`{'tabWidth': 4}`**
- Tabs: **false** 
  - @stylistic/no-tabs
  - @stylistic/no-mixed-spaces-and-tabs
- Semicolons: **true** 
  - @stylistic/semi
  - @stylistic/semi-style (for setting when false add semis at beginning of lines that may cause ASI failures)
- Quotes: **false** 
  - @stylistic/quotes
- Quote Props: **"as-needed"** 
  - @stylistic/quote-props
- JSX quotes: **false** 
  - @stylistic/jsx-quotes
- Trailing Commas: **"all"** 
  - @stylistic/comma-dongle
  - @stylistic/comma-style
- Bracket Spacing: **true** 
  - @stylistic/object-curly-spacing
- Bracket Line: **false** (May not be compatible with all file types in ESLint-style)
  - @stylistic/jsx-closing-bracket-location
- Arrow Function Parentheses: **always** 
  - @stylistic/arrow-parens

There are other options related to prettier config especially related to filetypes and parser which would require more digging into to see what are the related rules and how to set them.

**Comment by kevintyj** [comment]
Here is what the Prettier preferences would look like with ESLint-stylistic. I am still not 100% sure of how exactly indent and tabs would translate over:
_Some of the rules are just reiterating the default rules of eslint-stylistic, for the default behavior of eslint-stylistic refer to the previous comment_
```js
// .eslintrc.js

import * as pluginStylistic from '@stylistic/eslint-plugin'

export default {
  plugins: {
    // @antfu config style prefix rename
    style: pluginStylistic,
  },
  rules: {
    // more info on code comments in prettier is provided here: https://github.com/prettier/prettier/issues/265, hence it is disabled here by default
    // also, prettier's code wrap is different from eslint-stylistic's implementation. check: https://prettier.io/docs/en/options.html#print-width
    'style/max-len': ['error', { code: 80, tabWidth: 2 }],
    // line indentation for prettier check: https://www.emacswiki.org/emacs/SmartTabs
    'style/indent': ['error', 2],
    'style/semi': ['error', 'always'],
    'style/semi-style': ['error', 'last'],
    'style/quotes': ['error', 'double', { allowTemplateLiterals: true, avoidEscape: false }],
    'style/quote-props': ['error', 'as-needed'],
    'style/jsx-quotes': ['error', 'prefer-double'],
    // known issue with jsx comma dangle
    'style/comma-dangle': ['error', 'always-multiline'],
    'style/comma-style': ['error', 'last'],
    'style/object-curly-spacing': ['error', 'always'],
    // Prettier does not strictly enforce closing tag behavior rather ignores self closing tags
    'style/jsx-closing-bracket-location': ['error', {
      nonEmpty: 'tag-aligned',
      selfClosing: 'after-props',
    }],
    'style/arrow-parens': ['error', 'always'],
  },
}
```

**Comment by antfu** [comment]
Thanks for looking into it! I'd say we don't need to aim for 1:1 representation of Prettier (I guess for ppl who looking into this project would not use Prettier), but more like the "look and feel" of the general style and spacing of Prettier so ppl could have a nice looking foundation to fine-tune. So I'd say we probably don't need to include `style/max-len`. 

And also, we could have a factor function to accept high-level configuration (with the new flat config, and also possible to generate some static combinations for legacy extend format). Here is a reference in my personal config: https://github.com/antfu/eslint-config/blob/main/src/configs/stylistic.ts
